dnf/transaction_sr.py

```diff
@@ -220,25 +219,31 @@ def __init__(
         self._nevra_reason_cache = {}
         self._warnings = []
 
+    def load_from_file(self, fn):
+        self._filename = fn
         with open(fn, "r") as f:
             try:
-                self._replay_data = json.load(f)
+                replay_data = json.load(f)
             except json.decoder.JSONDecodeError as e:
                 raise TransactionFileError(fn, str(e) + ".")
 
         try:
-            self._verify_toplevel_json(self._replay_data)
+            self.load_from_dict(replay_data)
+        except TransactionError as e:
+            raise TransactionFileError(fn, e)
 
-            self._rpms = self._replay_data.get("rpms", [])
-            self._assert_type(self._rpms, list, "rpms", "array")
+    def load_from_dict(self, data):
+        self._replay_data = data
+        self._verify_toplevel_json(self._replay_data)
```

Comment:
The class (and whole transaction store/replay) was built with JSON as the data format, so this now becomes quite misleading, I think some renaming to make the method names sensible with the new usage of the data being just a structured dict would be good (the only reason to consider not cleaning it all up is dnf-5, but some renaming is cheap). Applies also to the `TransactionFileError`, which will now end up being thrown with an empty filename, but the message will no longer make any sense. So that should be certainly fixed (even though such errors would mean a problem in the code, it will be very misleading if it occurs).

Reply:
Let's do this in a different PR.